From 01cda63954dd7eb5468550f44c666eedc2706741 Mon Sep 17 00:00:00 2001
From: Ryan Eatmon <reatmon@ti.com>
Date: Wed, 12 Mar 2025 10:37:01 -0500
Subject: [PATCH] galUtil: Fix compile error

Fix several compile errors realted to incompatible pointers.  The fourth
argument is defined as void *.

galUtil.c: In function 'GalLoadDIBitmap':
galUtil.c:137:65: error: passing argument 4 of 'gcoOS_Open' from incompatible pointer type [-Wincompatible-pointer-types]
  137 |     if (gcmIS_ERROR(gcoOS_Open(gcvNULL, filename, gcvFILE_READ, &fp)))
      |                                                                 ^~~
      |                                                                 |
      |                                                                 FILE **

Upstream-Status: Inactive-Upstream [lastcommit: 2017-01-16]

Signed-off-by: Ryan Eatmon <reatmon@ti.com>
---
 .../hal/common/UnitTest/galUtil/galUtil.c     | 30 +++++++++----------
 1 file changed, 15 insertions(+), 15 deletions(-)

diff --git a/tests/src/test/hal/common/UnitTest/galUtil/galUtil.c b/tests/src/test/hal/common/UnitTest/galUtil/galUtil.c
index c2aefa6..f235f3a 100755
--- a/tests/src/test/hal/common/UnitTest/galUtil/galUtil.c
+++ b/tests/src/test/hal/common/UnitTest/galUtil/galUtil.c
@@ -134,7 +134,7 @@ unsigned char * CDECL GalLoadDIBitmap(const char *filename, BMPINFO **info)
     BMPFILEHEADER   header;       /* File header */
 
     /* Try opening the file; use "rb" mode to read this *binary* file. */
-    if (gcmIS_ERROR(gcoOS_Open(gcvNULL, filename, gcvFILE_READ, &fp)))
+    if (gcmIS_ERROR(gcoOS_Open(gcvNULL, filename, gcvFILE_READ, (void*) &fp)))
     {
         /* Failed to open the file. */
         GalOutput(GalOutputType_Error, "*ERROR*  Failed to open the file %s", filename);
@@ -335,7 +335,7 @@ int static GalSaveDIBitmap(const char *filename, BMPINFO *info, unsigned char *b
         return -1;
     }
 
-    if (gcmIS_ERROR(gcoOS_Open(gcvNULL, filename, gcvFILE_CREATE, &fp)))
+    if (gcmIS_ERROR(gcoOS_Open(gcvNULL, filename, gcvFILE_CREATE, (void*) &fp)))
     {
         /* Failed to open the file. */
         GalOutput(GalOutputType_Error, "*ERROR*  Failed to open the file %s", filename);
@@ -1927,7 +1927,7 @@ gctBOOL GalSaveSurfaceToVimg(gcoSURF surface, const char *FileName)
         head.imageStride = aStride = alignedWidth * bpp[0];
         lineSize = width[0] * bpp[0];
 
-        if (gcmIS_ERROR(gcoOS_Open(gcvNULL, FileName, gcvFILE_CREATE, &file)))
+        if (gcmIS_ERROR(gcoOS_Open(gcvNULL, FileName, gcvFILE_CREATE, (void*) &file)))
         {
             /* Failed to open the file. */
             GalOutput(GalOutputType_Error, "*ERROR*  Failed to open the file %s", FileName);
@@ -2070,7 +2070,7 @@ gceSTATUS GalSaveTSurfToVimg(
             gcmONERROR(gcvSTATUS_NOT_SUPPORTED);
         }
 
-        gcmONERROR(gcoOS_Open(NULL, Filename, gcvFILE_CREATE, &file));
+        gcmONERROR(gcoOS_Open(NULL, Filename, gcvFILE_CREATE, (void*) &file));
 
         gcmONERROR(gcoOS_Write(
             gcvNULL,
@@ -2808,7 +2808,7 @@ gceSTATUS CDECL GalLoadYUV2Surface2(
         gceSURF_FORMAT format;
         gctBOOL swap = gcvFALSE;
 
-        gcmONERROR(gcoOS_Open(Os, filename, gcvFILE_READ, &file));
+        gcmONERROR(gcoOS_Open(Os, filename, gcvFILE_READ, (void*) &file));
 
         head.Width  = read_dword(file);
         head.Height = read_dword(file);
@@ -3444,7 +3444,7 @@ gceSTATUS GalCreateTSurfWithPool(
         alignedBase = 256;
     }
 
-    gcmONERROR(gcoOS_Allocate(gcvNULL, sizeof(T2D_SURF), &surf));
+    gcmONERROR(gcoOS_Allocate(gcvNULL, sizeof(T2D_SURF), (void*) &surf));
     memset(surf, 0, sizeof(T2D_SURF));
 
     if (TileStatusConfig == gcv2D_TSC_2D_COMPRESSED ||
@@ -3869,7 +3869,7 @@ gceSTATUS GalLoadVimgToTSurfWithPool(
         gctUINT n;
         gctUINT32 vsupertile = ~0u;
 
-        gcmONERROR(gcoOS_Open(NULL, Filename, gcvFILE_READ, &file));
+        gcmONERROR(gcoOS_Open(NULL, Filename, gcvFILE_READ, (void*) &file));
 
         gcmONERROR(gcoOS_Read(
             NULL,
@@ -4092,7 +4092,7 @@ gceSTATUS CDECL GalLoadVimgToSurface(
         gceSURF_FORMAT format;
         gctBOOL swap = gcvFALSE;
 
-        gcmONERROR(gcoOS_Open(NULL, filename, gcvFILE_READ, &file));
+        gcmONERROR(gcoOS_Open(NULL, filename, gcvFILE_READ, (void*) &file));
 
         gcmONERROR(gcoOS_Read(
                 NULL,
@@ -4446,7 +4446,7 @@ gceSTATUS CDECL GalLoadVimgToSurfaceWithPool(
         gceSURF_FORMAT format;
         gctBOOL swap = gcvFALSE;
 
-        gcmONERROR(gcoOS_Open(NULL, filename, gcvFILE_READ, &file));
+        gcmONERROR(gcoOS_Open(NULL, filename, gcvFILE_READ, (void*) &file));
 
         gcmONERROR(gcoOS_Read(
                 NULL,
@@ -4800,7 +4800,7 @@ gceSTATUS CDECL GalLoadRawToSurface(
         gcoSURF surf;
         gctBOOL swap = gcvFALSE;
 
-        gcmONERROR(gcoOS_Open(NULL, filename, gcvFILE_READ, &file));
+        gcmONERROR(gcoOS_Open(NULL, filename, gcvFILE_READ, (void*) &file));
 
         /* Check the type. */
         switch (format)
@@ -5441,7 +5441,7 @@ gceSTATUS CDECL GalCreateMonoStream(
     GalMonoStream *stream = gcvNULL;
 
     do {
-        gcmERR_BREAK(gcoOS_Allocate(Os, sizeof(GalMonoStream), &stream));
+        gcmERR_BREAK(gcoOS_Allocate(Os, sizeof(GalMonoStream), (void*) &stream));
 
         memset(stream, 0, sizeof(GalMonoStream));
 
@@ -6196,7 +6196,7 @@ gceSTATUS CDECL GalLoadDECTPCRawToTSurfWithPool(
         ImageWidth, ImageHeight,
         Pool, &surf));
 
-    gcmONERROR(gcoOS_Open(NULL, DataName0, gcvFILE_READ, &file));
+    gcmONERROR(gcoOS_Open(NULL, DataName0, gcvFILE_READ, (void*) &file));
 
     aStride = surf->aWidth * bpp[0];
     lineSize = width[0] * bpp[0];
@@ -6247,7 +6247,7 @@ gceSTATUS CDECL GalLoadDECTPCRawToTSurfWithPool(
                 gcmONERROR(gcoOS_Close(gcvNULL, file));
             }
 
-            gcmONERROR(gcoOS_Open(NULL, DataName1, gcvFILE_READ, &file));
+            gcmONERROR(gcoOS_Open(NULL, DataName1, gcvFILE_READ, (void*) &file));
         }
 
         /* Copy data */
@@ -6304,7 +6304,7 @@ gceSTATUS CDECL GalLoadDECTPCRawToTSurfWithPool(
                     gcmONERROR(gcoOS_Close(gcvNULL, file));
                 }
 
-                gcmONERROR(gcoOS_Open(NULL, DataName2, gcvFILE_READ, &file));
+                gcmONERROR(gcoOS_Open(NULL, DataName2, gcvFILE_READ, (void*) &file));
             }
 
             /* Copy data */
@@ -6348,7 +6348,7 @@ gceSTATUS CDECL GalLoadDECTPCRawToTSurfWithPool(
     /* Tile status buffer */
     if (TableName0 != gcvNULL && surf->tileStatuslogical != gcvNULL)
     {
-        gcmONERROR(gcoOS_Open(NULL, TableName0, gcvFILE_READ, &file));
+        gcmONERROR(gcoOS_Open(NULL, TableName0, gcvFILE_READ, (void*) &file));
 
         /* Copy tile status data */
         for (n = 0; n < height[0]; n++)
@@ -6391,7 +6391,7 @@ gceSTATUS CDECL GalLoadDECTPCRawToTSurfWithPool(
             gctUINT aStride1 = (surf->aWidth/2) * bpp[1];
             gctUINT lineSize1 = width[1] * bpp[1];
 
-            gcmONERROR(gcoOS_Open(NULL, TableName1, gcvFILE_READ, &file));
+            gcmONERROR(gcoOS_Open(NULL, TableName1, gcvFILE_READ, (void*) &file));
 
             /* Copy tile status data */
             for (n = 0; n < height[1]; n++)
@@ -6443,7 +6443,7 @@ gceSTATUS CDECL GalLoadDECTPCRawToTSurfWithPool(
                 aStride2 = (surf->aWidth/2) * bpp[2];
                 lineSize2 = width[2] * bpp[2];
 
-                gcmONERROR(gcoOS_Open(NULL, TableName2, gcvFILE_READ, &file));
+                gcmONERROR(gcoOS_Open(NULL, TableName2, gcvFILE_READ, (void*) &file));
 
                 /* Copy tile status data */
                 for (n = 0; n < height[1]; n++)
-- 
2.17.1

